VAMobile/src/components/PhotoUpload.tsx

```diff
@@ -0,0 +1,166 @@
+  borderWidth: number
+  /** Hardcoded border color of white */
+  borderColor: string
+}
+
+const StyledImage = styled(Image)<StyledImageProps>`
+  width: ${themeFn<StyledImageProps>((_theme, props) => props.width)}px;
+  height: ${themeFn<StyledImageProps>((_theme, props) => props.height)}px;
+  border-radius: ${themeFn<StyledImageProps>((_theme, props) => props.borderRadius)}px;
+  border-width: ${themeFn<StyledImageProps>((_theme, props) => props.borderWidth)}px;
+  border-color: ${themeFn<StyledImageProps>((_theme, props) => props.borderColor)};
+`
+//TODO: Add this back to the VeteranStatusScreen when decided upon
+const PhotoUpload: FC<PhotoUploadProps> = ({ width, height }) => {
+  const { t } = useTranslation(NAMESPACE.COMMON)
+  const confirmAlert = useDestructiveActionSheet()
+  const photoUploadBorderRadius = 50
+  const photoUploadBorderWidth = 2
+  const showActionSheetWithOptions = useShowActionSheet()
+  const VETERAN_STATUS_PHOTO_KEY = '@store_veteran_status_photo'
+  const [uri, setUri] = useState('')
+  const options = [t('fileUpload.camera'), t('fileUpload.photoGallery'), t('cancel')]
+  const uploadBorderColor = theme.colors.border.photoUpload
+
+  const getPhotoFromStorage = async (): Promise<void> => {
+    const storedUri = await AsyncStorage.getItem(VETERAN_STATUS_PHOTO_KEY)
+    if (storedUri) {
+      setUri(storedUri)
+    }
+  }
+
+  useEffect(() => {
+    if (uri === '') {
+      getPhotoFromStorage()
+    }
+  }, [uri])
+
+  const photo = (
+    <StyledImage source={{ uri }} width={width} height={height} borderRadius={photoUploadBorderRadius} borderWidth={photoUploadBorderWidth} borderColor={uploadBorderColor} />
+  )
+
+  const uploadCallback = (response: ImagePickerResponse): void => {
+    const { assets, errorMessage, didCancel } = response
+    const picture = assets?.[0]?.uri
+    if (didCancel) {
+      return
+    } else if (errorMessage) {
+      //TODO Snackbar in this case? Discuss error cases
+      return
+    } else if (picture) {
+      setUri(picture)
+      AsyncStorage.setItem(VETERAN_STATUS_PHOTO_KEY, picture)
+    } else {
+      return
+    }
+  }
+
+  const onPress = (): void => {
+    if (uri) {
+      confirmAlert({
+        title: t('removePhoto'),
+        cancelButtonIndex: 0,
+        destructiveButtonIndex: 1,
+        buttons: [
+          {
+            text: t('keep'),
+            onPress: () => {},
+          },
+          {
+            text: t('remove'),
+            onPress: () => {
+              setUri('')
+              AsyncStorage.setItem(VETERAN_STATUS_PHOTO_KEY, '')
+            },
+          },
+        ],
+      })
+    } else {
+      showActionSheetWithOptions(
+        {
+          options,
+          cancelButtonIndex: 2,
+        },
+        (buttonIndex) => {
+          switch (buttonIndex) {
+            case 0:
+              launchCamera({ mediaType: 'photo', quality: 0.9, includeBase64: true }, (response: ImagePickerResponse): void => {
```

Comment:
Should any of these props be constants? Same for launchImageLibrary below

Reply:
Probably, but it'd need to be app wide not just here (I copied these values from elsewhere).